lib/galaxy/files/sources/webdav.py

```diff
@@ -22,6 +23,11 @@ class WebDavFilesSource(PyFilesystem2FilesSource):
 
     def _open_fs(self, user_context=None, opts: Optional[FilesSourceOptions] = None):
         props = self._serialization_props(user_context)
+        use_temp_files = props.get("use_temp_files")
+        if use_temp_files is None:
+            # Default to True to avoid memory issues with large files.
+            props["use_temp_files"] = True
+            props["temp_path"] = props.get("temp_path", tempfile.gettempdir())
```

Comment:
I would give it a clean directory with `tempfile.TemporaryDirectory(prefix="webdav_")` ?
Are the created files going to be cleaned up ?

Reply:
> Are the created files going to be cleaned up ?

That's a good question, I need to check what the internal fs plugin does...

Reply:
It seems none of the file source plugins explicitly clean up after downloading or uploading files as far as I can tell.

Reply:
Alright, well, admins should already have something in place to clean up temp dirs. It might be nice in the future, but this is fine for now.